crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -166,29 +186,38 @@ impl Object {
             .extend()
     }
 
-    /// Objects can either be immutable, shared, owned by an address,
-    /// or are child objects (part of a dynamic field)
-    async fn kind(&self) -> Option<ObjectKind> {
+    /// The owner type of this Object.
+    /// Immutable and Shared Objects do not have owners.
+    async fn owner(&self, ctx: &Context<'_>) -> Option<ObjectOwnerNew> {
         use NativeOwner as O;
-        use ObjectKind as K;
-        Some(match self.native.owner {
-            O::AddressOwner(_) => K::Owned,
-            O::ObjectOwner(_) => K::Child,
-            O::Shared { .. } => K::Shared,
-            O::Immutable => K::Immutable,
-        })
-    }
 
-    /// The Address or Object that owns this Object.  Immutable and Shared Objects do not have
-    /// owners.
-    async fn owner(&self) -> Option<Owner> {
-        use NativeOwner as O;
-        let (O::AddressOwner(address) | O::ObjectOwner(address)) = self.native.owner else {
-            return None;
-        };
+        if self.native.owner.is_child_object() {
+            let parent = match self.native.owner.get_owner_address().ok() {
+                Some(addr) => ctx
+                    .data_unchecked::<PgManager>()
+                    .fetch_obj(addr.into(), None)
+                    .await
+                    .ok()
+                    .flatten(),
+                None => None,
+            };
+            return Some(ObjectOwnerNew::Parent(Parent { parent }));
+        }
 
-        let address = SuiAddress::from(address);
-        Some(Owner { address })
+        match self.native.owner {
+            O::AddressOwner(address) | O::ObjectOwner(address) => {
+                let address = SuiAddress::from(address);
+                Some(ObjectOwnerNew::Address(AddressOwner {
+                    owner: Some(Owner { address }),
+                }))
+            }
+            O::Shared {
+                initial_shared_version,
+            } => Some(ObjectOwnerNew::Shared(Shared {
+                initial_shared_version: initial_shared_version.value(),
+            })),
+            O::Immutable => Some(ObjectOwnerNew::Immutable(Immutable { dummy: Some(true) })),
```

Comment:
The schema is as following, so not sure if this is the right way.
```
type Immutable {
  # Dummy field
  _: Boolean
}
```

Reply:
Ah, this is what you can do:
struct GasCoin {
    /// A workaround to define an empty variant of a GraphQL union.
    #[graphql(name = "_")]
    dummy: Option<bool>,
}

lifted from programmble.rs in the sui-graphql-rpc code

Reply:
It's correct, but I would make the dummy field have a `None` value so it doesn't actually show up with the strange field name, in the response.